pennylane/_device.py

```diff
@@ -729,6 +729,10 @@ def batch_transform(self, circuit):
                     "Can only return the expectation of a single Hamiltonian observable"
                 ) from e
 
+        if len(circuit._obs_sharing_wires) > 0 and not hamiltonian_in_obs:
+            # Check for case of non-commuting terms and that there are no Hamiltonians
+            return qml.transforms.split_non_commuting(circuit)
```

Comment:
I suppose though this won't allow

```python
return expval(H), qml.PauliX(0)
```

_But_ I think this is out of scope, and something for another ticket/PR if we decide this should be supported :)

Reply:
I think it would be quite useful to have Hamiltonians accepted as well since anything `scalar * obs` is treated as a Hamiltonian, but it would indeed require some more thorough refinements that might be worth a new PR. I will add a TODO marker here in the code.